git-branchless/src/commands/query.rs

```diff
@@ -49,12 +49,13 @@ pub fn query(
             let (effects, _progress) = effects.start_operation(OperationType::SortCommits);
             let _effects = effects;
 
-            let commit_set = dag.query().sort(&commit_set)?;
             let commit_set = commit_set.intersection(&dag.branch_commits);
-            commit_set_to_vec_unsorted(&commit_set)?
+            let commit_set = dag.query().sort(&commit_set)?;
```

Comment:
Seemed better to sort after intersecting, just in case intersection doesn't preserve order